sample-app/src/components/DropdownSection.tsx

```diff
@@ -94,7 +94,7 @@ export default function DropdownSection({
         key={index}
         className={className}
         id={`${optionIdPrefix}-${index}`}
-        onClick={() => onClickOption(option, index)}>
+        onClick={() => onSubmit(index, option)}>
```

Comment:
Yeah, that makes sense. Since the option is optional for this function while the index is required, I get an error if I have the param ordering switched. So, I can switch the ordering for renderOption to have it be consistent

Reply:
Done!

Reply:
Can you explain why index is required and option is optional?

Reply:
When an option is submitted, FilterSearch would always need to know what the index of the option was to determine which filter to select. But, the option is only needed if we need to access its value to set the text in the input bar. When navigating to the option using arrow keys and selecting with Enter, the text in the input bar has already been set to the correct value and does not need to be set again. However, if an option is clicked, the text in the input bar needs to be set. By making option optional, we can avoid an extra call to set the input text.

But, if it's confusing to have it optional, we could make them both required and just always set the input text on submission, even if it isn't needed for submitting with Enter and would result in an extra render.

Reply:
I think that we should design the `DropdownSection` in a way that makes the most sense for itself. I would prefer to alway include the option and we can have the `FilterSearch` component be responsible for preventing extra renders if needed. For the scope of this item, I think an extra render is okay.

Reply:
By making `option` required, we can swap `option` and `index`. I see option as the primary piece of data, and index as more of metadata, so putting option first makes more sense. Also, I'm not sure we need the render function of the option in the callback, so maybe we could use the options's `value` instead?

Reply:
That makes sense. I'll change it to pass `(optionValue, index)`

Reply:
Updated to `onSelect(optionValue, optionIndex)`